src/format/typst/format-typst.ts

```diff
@@ -103,3 +107,21 @@ export function typstFormat(): Format {
     },
   });
 }
+
+function typstResolveFormat(format: Format) {
+  // Pandoc citeproc with typst output requires adjustment
+  // https://github.com/jgm/pandoc/commit/e89a3edf24a025d5bb0fe8c4c7a8e6e0208fa846
+  if (format.pandoc?.[kCiteproc] === false) {
+    // TODO: Patch to do what Pandoc > 3.1.2 will do next when `+citations` (the default) is set.
+    format.pandoc[kVariables] = format.pandoc[kVariables] || {};
+    format.pandoc[kVariables]["citations"] = true;
+  } else if (
+    format.pandoc?.[kCiteproc] === true || format.metadata?.[kBibliography] ||
+    format.metadata?.[kReferences]
+  ) {
```

Comment:
Using same logic as in `CiteMethod()` to know if we are running `citeproc`. 

We can't do just  `} else {` as we are using `resolveFormats()` which apply on user metadata and not default format metadata (where we could have set `citeproc: true`). 

We resolve this 3 times (projFormats, directoryFormats, inputFormats), and a metadata can be unset in one or several of those cases. So we need to be explicit